traffic_portal/app/src/modules/public/sso/SsoController.js

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var SsoController = function($scope, $state, $interval, $location, authService, propertiesModel) {
+
+	var init = function () {
+        const tokenParameter = propertiesModel.properties.oAuth.oAuthTokenQueryParam !== '' ? propertiesModel.properties.oAuth.oAuthTokenQueryParam : 'access_token';
+		const token = $location.search()[tokenParameter];
```

Comment:
Looking at this implementation, you seem to be setting up an "Implicit Grant" ([RFC6749 Section 4.2](https://tools.ietf.org/html/rfc6749#section-4.2)) type of OAuth. However, the Implicit Grant specifies that the [Access Token Response](https://tools.ietf.org/html/rfc6749#section-4.2.2) will be found within parameters in

> _"the fragment component of the redirection URI using the 'application/x-www-form-urlencoded' format"_

Whereas [the documentation of `$location.search`](https://docs.angularjs.org/api/ng/service/$location#search) indicates it will pull out the "search part" (example shows that this is the query string) of a URL - **not** the fragment. So will this work for OAuth2-compliant servers?